libraries/chain/include/steem/chain/steem_object_types.hpp

```diff
@@ -203,6 +203,69 @@ namespace fc
       {
          s.read( (char*)&id._id, sizeof(id._id));
       }
+
+      template< typename Stream >
+      inline void pack( Stream& s, const chainbase::shared_string& ss )
+      {
+         std::string str = steem::chain::to_string( ss );
+         fc::raw::pack( s, str );
+      }
+
+      template< typename Stream >
+      inline void unpack( Stream& s, chainbase::shared_string& ss )
+      {
+         std::string str;
+         fc::raw::unpack( s, str );
+         steem::chain::from_string( ss, str );
+      }
+
+      template< typename Stream, typename E, typename A >
+      inline void pack( Stream& s, const boost::interprocess::deque<E, A>& dq )
+      {
+         // This could be optimized
+         std::vector<E> temp;
+         std::copy( dq.begin(), dq.end(), std::back_inserter(temp) );
+         pack( s, temp );
+      }
+
+      template< typename Stream, typename E, typename A >
+      inline void unpack( Stream& s, boost::interprocess::deque<E, A>& dq )
+      {
+         // This could be optimized
+         std::vector<E> temp;
+         unpack( s, temp );
+         std::copy( temp.begin(), temp.end(), std::back_inserter(dq) );
+      }
+
+      template< typename Stream, typename K, typename V, typename C, typename A >
+      inline void pack( Stream& s, const boost::interprocess::flat_map< K, V, C, A >& value )
+      {
+         fc::raw::pack( s, unsigned_int((uint32_t)value.size()) );
+         auto itr = value.begin();
+         auto end = value.end();
+         while( itr != end )
+         {
+           fc::raw::pack( s, *itr );
```

Comment:
This looks copied directly from FC's map pack, which is great (fewer opportunities for error), but it copied over the inconsistent indentation of FC as well... The internals of the while loop are only indented twice.